server/src/main/java/org/opensearch/index/translog/transfer/TranslogTransferManager.java

```diff
@@ -156,7 +156,7 @@ public boolean transferSnapshot(TransferSnapshot transferSnapshot, TranslogTrans
 
             try {
                 if (latch.await(TRANSFER_TIMEOUT_IN_MILLIS, TimeUnit.MILLISECONDS) == false) {
-                    Exception ex = new TimeoutException("Timed out waiting for transfer of snapshot " + transferSnapshot + " to complete");
+                    Exception ex = new TranslogUploadFailedException("Timed out waiting for transfer of snapshot " + transferSnapshot + " to complete");
```

Comment:
can we add an IT for same ?

Reply:
Incorporated as UTs. It is a bit tricky to simulate the timeout in ITs.